### PR TITLE
chore: Switch from edx-sphinx-theme to sphinx-book-theme

### PR DESCRIPTION
**Pull request by xitij2000**
The edx-sphinx theme is being deprecated, and replaced with sphinx-book-theme. This removes references to the deprecated theme and replaces them with the new standard theme for the platform.

**Testing instructions:** 
- Run `make docs` and see that the docs are generated properly and use the sphinx-book-theme

See https://github.com/openedx/public-engineering/issues/200
----

I've completed each of the following or determined they are not applicable:

- [x] Made a plan to communicate any major developer interface changes (or N/A)

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @xitij2000!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by itsjeyd** [comment]
Hi @xitij2000, just a heads-up that I'm marking this as waiting for author since internal review hasn't happened yet.

Will update status again when it's ready for upstream review.

**Comment by Cup0fCoffee** [comment]
:+1:

- [x] I tested this: ran `make docs`
- [x] I read through the code

cc. @itsjeyd

File: docs/conf.py [hunk 1]
```diff
@@ -158,16 +158,46 @@
...
+                src="https://i.creativecommons.org/l/by-sa/4.0/80x15.png"/>
+        </a>
+        <br>
+        These works by
+            <a
+                xmlns:cc="https://creativecommons.org/ns#"
+                href="https://openedx.org"
+                property="cc:attributionName"
+                rel="cc:attributionURL"
+            >The Center for Reimagining Learning</a>
```
Comment (feanil) [inline docs/conf.py]:
```suggestion
            >The Axim Collaborative</a>
```

**Review by feanil — APPROVED** [review]
One smal change and then I think this is good to go.

**Comment by feanil** [comment]
The only failing issue is related to the missing codecov package so I think this PR is good to merge as is.

**Comment by openedx-webhooks** [comment]
@xitij2000 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/devstack&entry.867055334=https://github.com/openedx/devstack/pull/1041&entry.1484655318=https://github.com/xitij2000&entry.752974735=2023-03-24+07:30&entry.1917517419=2023-04-14+15:49&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by itsjeyd** [comment]
label: core contributor

**Pull request merged**